libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1674,8 +1674,8 @@ bool TypeChecker::visit(Identifier const& _identifier)
 	if (auto variableDeclaration = dynamic_cast<VariableDeclaration const*>(annotation.referencedDeclaration))
 		annotation.isPure = annotation.isConstant = variableDeclaration->isConstant();
 	else if (dynamic_cast<MagicVariableDeclaration const*>(annotation.referencedDeclaration))
-		if (auto functionType = dynamic_cast<FunctionType const*>(annotation.type.get()))
-			annotation.isPure = functionType->isPure();
+		if (dynamic_cast<FunctionType const*>(annotation.type.get()))
+			annotation.isPure = true;
```

Comment:
I don't think positives or negatives are relevant.  I'm seeing only booleans.  If an expression reads, and if an expression writes.

We can also do this in the follow-up.  In that case, I'll add this point in the follow-up issue.